### PR TITLE
Feat/manage pictures

### PR DESCRIPTION
**Pull request by Aurelien9Code**
This PR creates a picture selection tool and integrate it in the AccountDetails. 
The tool supports selecting multiple images with a maximum limit and provides options to validate or cancel the selection.

The composable displays selected images in a grid and enables the removal of individual selections, handles validation to confirm the selection and cancellation to reset the selection process, supports a dynamic maximum limit for the number of pictures to be selected.

It also uses the ModalBottomSheet to allow smooth and interactive image selection, as this creates a "pop-up" page that can be easily cancelled by clicking anywhere else on the screen.

### CODE REVIEW

File: app/src/androidTest/java/com/android/unio/end2end/UserAccountCreationTest.kt [hunk 1]
```diff
@@ -85,6 +85,11 @@ class UserAccountCreationTest : EndToEndTest() {
     composeTestRule.onNodeWithTag(InterestsOverlayTestTags.SAVE_BUTTON).assertIsDisplayed()
     composeTestRule.onNodeWithTag(InterestsOverlayTestTags.SAVE_BUTTON).performClick()
 
+    composeTestRule.onNodeWithTag(AccountDetailsTestTags.PROFILE_PICTURE_ICON).assertIsDisplayed()
+    composeTestRule.onNodeWithTag(AccountDetailsTestTags.PROFILE_PICTURE_ICON).performClick()
+
+    // balablablaGoBack !dsajdsajdbsakjdsa
```
Comment (Redd87) [inline app/src/androidTest/java/com/android/unio/end2end/UserAccountCreationTest.kt:91]: ?

File: app/src/androidTest/java/com/android/unio/components/authentication/PictureSelectionToolTest.kt [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+import dagger.hilt.android.testing.HiltAndroidTest
+import io.mockk.MockKAnnotations
+import io.mockk.every
+import io.mockk.impl.annotations.MockK
+import io.mockk.mockk
+import io.mockk.mockkStatic
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.mockito.Mockito.mock
```
Comment (Redd87) [inline app/src/androidTest/java/com/android/unio/components/authentication/PictureSelectionToolTest.kt:25]: Watch out mixing mockk and mockito and try to stick to only mockk

**Review by Redd87 — CHANGES_REQUESTED** [review]
Good job overall! Just translate the few mockito references to mockk and it's good to me

File: app/src/androidTest/java/com/android/unio/components/authentication/PictureSelectionToolTest.kt (hunk 2 above)
Comment (Aurelien9Code) [reply app/src/androidTest/java/com/android/unio/components/authentication/PictureSelectionToolTest.kt:25]: sure !

File: app/src/androidTest/java/com/android/unio/end2end/UserAccountCreationTest.kt (hunk 1 above)
Comment (Aurelien9Code) [reply app/src/androidTest/java/com/android/unio/end2end/UserAccountCreationTest.kt:91]: oups haha

**Review by Redd87 — APPROVED** [review]
As I'm going to sleep I trust you to make the changes correctly

File: app/src/main/java/com/android/unio/ui/components/PictureSelectionTool.kt [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+   * Creates a temporary URI for storing an image taken by the camera.
+   *
+   * @return A URI pointing to the location where the new image will be stored.
+   */
+  fun createImageUri(): Uri {
+    val resolver = context.contentResolver
+    val contentValues =
+        ContentValues().apply {
+          put(MediaStore.Images.Media.DISPLAY_NAME, "new_profile_picture.jpg")
+          put(MediaStore.Images.Media.MIME_TYPE, "image/jpeg")
```
[the remark this answers is not in the document]
Comment (Aurelien9Code) [reply app/src/main/java/com/android/unio/ui/components/PictureSelectionTool.kt:88]: As they were really arbitrary and never used anywhere else, i thought it did not need to be in another place
But I will create a const val

**Comment by Aurelien9Code** [comment]
> As I'm going to sleep I trust you to make the changes correctly

I did make them : )

**Pull request merged**